pfcpiface/parse-pdr.go

```diff
@@ -244,35 +486,26 @@ func (p *pdr) parsePDI(seid uint64, pdiIEs []*ie.IE, appPFDs map[string]appPFD,
 				p.appFilter.dstIPMask = ipMask2int(ipf.dst.IPNet.Mask)
 				p.appFilter.srcIP = ip2int(ipf.src.IPNet.IP)
 				p.appFilter.srcIPMask = ipMask2int(ipf.src.IPNet.Mask)
-
-				if ipf.src.Port > 0 {
-					p.appFilter.srcPort, p.appFilter.srcPortMask = ipf.src.Port, 0xffff
-				}
-
-				if ipf.dst.Port > 0 {
-					p.appFilter.dstPort, p.appFilter.dstPortMask = ipf.dst.Port, 0xffff
-				}
+				p.appFilter.dstPortFilter = ipf.dst.ports
+				p.appFilter.srcPortFilter = ipf.src.ports
 
 				// FIXME: temporary workaround for SDF Filter,
 				//  remove once we meet spec compliance
-				p.appFilter.srcPort = p.appFilter.dstPort
-				p.appFilter.dstPort, p.appFilter.dstPortMask = 0, 0 // reset UE Port
+				p.appFilter.srcPortFilter = p.appFilter.dstPortFilter
+				p.appFilter.dstPortFilter = newWildcardPortFilter()
 			} else if p.srcIface == access {
 				p.appFilter.srcIP = ip2int(ipf.dst.IPNet.IP)
 				p.appFilter.srcIPMask = ipMask2int(ipf.dst.IPNet.Mask)
 				p.appFilter.dstIP = ip2int(ipf.src.IPNet.IP)
 				p.appFilter.dstIPMask = ipMask2int(ipf.src.IPNet.Mask)
-				if ipf.dst.Port > 0 {
-					p.appFilter.srcPort, p.appFilter.srcPortMask = ipf.dst.Port, 0xffff
-				}
-				if ipf.src.Port > 0 {
-					p.appFilter.dstPort, p.appFilter.dstPortMask = ipf.src.Port, 0xffff
-				}
+				// Ports are flipped for access PDRs
+				p.appFilter.dstPortFilter = ipf.src.ports
+				p.appFilter.srcPortFilter = ipf.dst.ports
```

Comment:
@osinstom Can you verify this?

Reply:
Yes, this is how we parse SDF Filter because we are only interested in application IP/port. I guess it should work for BESS too.